### PR TITLE
Finish chapter 9

### PR DESCRIPTION
**Pull request by phongnm-0776**
Ready
image

### CODE REVIEW

**Comment by phongnm-0776** [comment]
Ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -4,16 +4,50 @@ def log_in user
...
+  end
+
+  def login_user params
+    if @user && @user.authenticate(params[:session][:password])
+      log_in @user
+      params[:session][:remember_me] == "1" ? remember(@user) : forget(@user)
+      redirect_to @user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
+      render "new"
```
Comment (chien-pv) [inline app/helpers/sessions_helper.rb]: :new

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -4,16 +4,50 @@ def log_in user
...
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def login_user params
+    if @user && @user.authenticate(params[:session][:password])
+      log_in @user
+      params[:session][:remember_me] == "1" ? remember(@user) : forget(@user)
+      redirect_to @user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
```
Comment (chien-pv) [inline app/helpers/sessions_helper.rb]: Dùng I18n nhé em

**Comment by chien-pv** [comment]
commented

**Comment by phongnm-0776** [comment]
ready

**Comment by chien-pv** [comment]
merged

**Pull request merged**